impeller/renderer/backend/vulkan/render_pass_vk.cc

```diff
@@ -307,26 +307,18 @@ SharedHandleVK<vk::Framebuffer> RenderPassVK::CreateVKFramebuffer(
 // |RenderPass|
 void RenderPassVK::SetPipeline(
     const std::shared_ptr<Pipeline<PipelineDescriptor>>& pipeline) {
-  PipelineVK& pipeline_vk = PipelineVK::Cast(*pipeline);
+  pipeline_ = pipeline;
```

Comment:
std::shared_ptr release is going to show up traces for apps with lots of drawing commands with this change.

I think that we should adjust this API to accept const Pipeline<PipelineDescriptor>>* or something else, given that the variants maps and/or caches already keep the pipelines alive

Reply:
Or make `Draw` take a `pipeline`? We can keep tinkering on this API for performance. I was just looking to make things work TBH.

Reply:
That sounds like a good idea. No action necessary here as this issue is relatively minor compared to the importance of this patch.